### DISCUSSION TITLE
How to select multiple adjacent nodes of the same level at once?

### DISCUSSION THREAD

**Discussion by ddsquires** [freeplane/freeplane#677] [Help]
Hello, is it possible to select multiple adjacent nodes of the same level at once? I see ctrl+clicking each works as I would expect it to, but I'm looking for something more like shift+click or drawing a box around a given area to select a bunch. 

Ultimately I want to be able to fold/unfold all of the nodes downstream of what is selected at once, or even change the formatting together without doing a level style formatting or having to ctrl+click each node individually.

Thanks.

**Comment by euu2021** [comment]
Please, take a look at [this](https://sourceforge.net/p/freeplane/featurerequests/2578/). I'm not sure if that's what you want. 

If it's not enough for you, it's possible to think of a script that makes the operation easier.

**Comment by sifran-github** [comment]
I also support this request. I think that shift clicking can be implemented without too much programming complications.

What i sometimes use right now is a custom script, select children. You select the parent node (or nodes) and run the script to move the selection to the children.

Put this in the [freeplane user directory]\scripts as select_children.groovy:

```
// @ExecutionModes({ON_SINGLE_NODE})
def toSelect = c.selecteds.collect{ node -> node.find {it.parent == node} }.sum()
c.selectMultipleNodes(toSelect)
```

**Comment by sifran-github** [comment]
another tip: if you just want to fold or unfold the children of the currently selected nodes, use the built in keyboard shortcut... ctrl+space.

**Comment by EdoFro** [comment]
you can also use `shift + NextPage` to select nodes of the same level, that are adjacent and have different parents

**Reply by euu2021** [reply]
(NextPage in my laptop is called  PageUp/PageDown)

Wow, I didn't know this. Nice!

**Reply by sifran-github** [reply]
interesting, it does help with selecting. I didn't know about it as well.

But just notice that if you keep on clicking shift+pagedown, the selection can jump to nodes on a different level.
probably this shortcut's logic is: "select nodes on the same level, but if these nodes do not exist, select the closest parent level".

**Reply by euu2021** [reply]
Yes, it's the same logic of the PageUp/PageDown, when they are used without the Shift.

**Reply by ddsquires** [reply]
Thanks @EdoFro, this is pretty close to what I'm looking for. Some notes:

1. On my Windows laptop, shift+PageDown ("PgDn") only captures 5 nodes at the same level at a time. Continuing to do this captures 5 at a time. So If I have a couple dozen, it's a handful of Shift+PgDn+PgDn+PgDn..... OR Shift+HOLD PgDn to get to the bottom.
2. My compact external keyboard only offers PgDn as a function of the down arrow, so the key sequence is Fn+Shift+PgDn (and NOT Shift+Fn+PgDn).
3. Alternatively, Shift+Down arrow repeatedly or holding the down arrow key gets the job done too, just a little less efficiently. Best option for smaller sets of nodes or fine tuning.
4. Side note from exploring this a bit further, Shift+left or right arrow has some interesting behaviors that I haven't dove into. something like selecting within the branch but then when you reach a parent that has children at other branches it automatically selects all children of all levels. not just the branch you originated from.

My apologies of any of those notes are better known, I'm new here!

**Reply by sifran-github** [reply]
@ddsquires i think the logic of shift+left (on a right side of the root) is "choose the parent, and apply shift+right from there" which is basically "choose all visible nodes under current node"

**Comment by autoca** [comment]
You might also like to look at macmarrum's freeplane scripts starting with "select.." or "selection..", e.g. "selectSiblingsAndMyself.groovy":
https://github.com/macmarrum/freeplane-scripting/tree/main/src/scripts

Also, note his remark here about adapting the placement of the script in the menu: https://github.com/freeplane/freeplane/discussions/607#discussioncomment-3509547

Of course, his other scripts are also interesting.
